ludwig/features/text_feature.py

```diff
@@ -106,17 +106,29 @@ def get_feature_meta(
             padding_symbol,
             unknown_symbol,
         ) = tf_meta
-        # Use max_sequence_length if provided, otherwise use max length found in dataset.
-        if preprocessing_parameters["max_sequence_length"] is not None:
-            logger.info("Using max_sequence_length provided in preprocessing parameters")
-            max_sequence_length = preprocessing_parameters["max_sequence_length"]
-            max_sequence_length_99ptile = max_sequence_length
+
+        # Use sequence_length if provided, otherwise use max length found in dataset.
+        if preprocessing_parameters["sequence_length"] is not None:
+            logger.info(
+                f"Setting max length to sequence_length={preprocessing_parameters['sequence_length']} provided in "
+                f"preprocessing parameters"
+            )
+            max_sequence_length = preprocessing_parameters["sequence_length"]
+            max_sequence_length_99ptile = preprocessing_parameters["sequence_length"]
         else:
-            logger.info("Inferring max_sequence_length from dataset")
             max_sequence_length = max_len + 2  # For start and stop symbols.
             max_sequence_length_99ptile = max_len_99ptile + 2  # For start and stop symbols.
-        logger.info(f"Using max sequence length of {max_sequence_length} for feature '{column.name}'")
+            logger.info(f"Setting max length using dataset: {max_sequence_length} (including start and stop symbols)")
+
+            if preprocessing_parameters["max_sequence_length"] < max_sequence_length:
```

Comment:
I believe that this will fail `if preprocessing_parameters["sequence_length"] is None and preprocessing_parameters["max_sequence_length"] is None`. Perhaps we could add an aux check for that too.

Reply:
Good point! Conceptually, the case where they are both None just means that we infer from the dataset entirely (without setting a hard cap on the text length). I'll be sure this is reflected in the code (you are right that this would fail as-is).